go-controller/pkg/node/gateway_localnet.go

```diff
@@ -218,7 +223,9 @@ func initLocalnetGateway(nodeName string, subnet *net.IPNet, wf *factory.WatchFa
 	}
 
 	if config.Gateway.NodeportEnable {
-		err = localnetNodePortWatcher(ipt, wf, gatewayIP)
+		err = n.localnetNodePortWatcher(
```

Comment:
This line reads as just... kind of awkward. Can this be consolidated? Something like `n.watchLocalPorts()` that sets up the handlers?

Reply:
I've re-worked this part in my update, might be a bit better now that I've split out the "port claiming" part to its own watcher. Anyhow, I did it like that as to be able to write tests easier.